### PR TITLE
Add automated testing for OSX

### PR DESCRIPTION
**Pull request by pimterry**
Connects to https://github.com/resin-io/resin-cli/issues/542.

This does pass on Travis, happily, but it's marked as failed because of AppVeyor: that's because I haven't set up Windows builds yet. It will work in time (I've got the config, but the tests don't actually pass on windows), but not yet. For now I've set AppVeyor to ignore builds without appveyor.yml, so there shouldn't be any noise from these in future until that's actually added.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -1,8 +1,11 @@
 
 language: node_js
+os:
+  - linux
+  - osx
 node_js:
-  - "5"
+  - "8"
```
Comment (lekkas) [inline .travis.yml]: Is `8` supported in travis yet? - https://docs.travis-ci.com/user/languages/javascript-with-nodejs/

Comment (pimterry) [inline .travis.yml]: Huh, interesting! Certainly seems to work: https://travis-ci.org/resin-io-modules/resin-discoverable-services/jobs/253118857 :smile:

Looks like their docs are out of date: https://github.com/travis-ci/docs-travis-ci-com/pull/1278

Comment (lekkas) [inline .travis.yml]: Ah, lovely, so it is supported after all. I remember bumping into this a few weeks before (can't remember exactly where) and apparently `8` was not supported back then.

**Comment by pimterry** [comment]
Oooh, interesting problem: the OSX tests are a little unstable, because their running on the same network as one another, and are returning `Service name is already in use on the network` errors.

I'm going to update the tests to come up with unique service names, so you can run them in parallel on the same network.

**Pull request merged**